packages/dart_frog_cli/lib/src/commands/list/list.dart

```diff
@@ -0,0 +1,75 @@
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:dart_frog_cli/src/command.dart';
+import 'package:dart_frog_gen/dart_frog_gen.dart';
+import 'package:mason/mason.dart';
+
+/// Definition for a function that builds a [RouteConfiguration] from a
+/// [Directory].
+typedef RouteConfigurationBuilder = RouteConfiguration Function(
+  Directory directory,
+);
+
+/// {@template list_command}
+/// `dart_frog list "path/to/project"`
+///
+/// Lists the routes on the project.
+/// {@endtemplate}
+class ListCommand extends DartFrogCommand {
+  /// {@macro list_command}
+  ListCommand({
+    super.logger,
+    RouteConfigurationBuilder buildConfiguration = buildRouteConfiguration,
+  }) : _buildConfiguration = buildConfiguration;
+
+  final RouteConfigurationBuilder _buildConfiguration;
+
+  @override
+  String get description => 'Lists the routes on a Dart Frog project.';
+
+  @override
+  String get name => 'list';
+
+  @override
+  final String invocation = 'dart_frog list "path/to/project"';
+
+  @override
+  Future<int> run() async {
+    final projectDir = _projectDirectory;
+
+    final configuration = _buildConfiguration(projectDir);
+
+    logger
+      ..info('Route list 🐸:')
+      ..info('==============\n');
```

Comment:
Nit: Remove header to simplify grep operations?

```suggestion
```

Perhaps for pretty prints, we can introduce add a subcommand to pretty print routes in a tree structure or table (in another PR, if proposal is agreed by the team or community).

Reply:
I like the idea of having a grep friendly output. But I feel that this would mostly be meant by human eyes, so I would argue that this should be the default.

I am gonna go ahead and implement a flag for this "cleaner" output.

And for the additional info, such as total number, etc, we sure can do in a follow up PR :)